program.sh

```diff
@@ -1,17 +1,47 @@
 #!/bin/sh
 
+# 10 min interval
+INTERVAL=600
+STOP=stop
+
 download_repository() {
-	echo "swift download $1"
-	swift download $1
+    echo "swift download $1"
+    swift download $1
 }
 
 upload_repository() {
-	swift upload $1 repository/ sync/
+    swift upload $1 repository/ sync/
 } 
 
-trap 'kill -TERM $PID; wait $PID; upload_repository $1; exit 143' TERM
+watch_upload() {
+    trap 'kill -TERM $PIDSLEEP' SIGTERM
+
+    while ! [ -f $STOP ]
+    do
+	sleep $INTERVAL &
+	PIDSLEEP=$!
+	wait $PIDSLEEP
+	date
+	upload_repository $1
+    done
+}
+
+before_exit() {
+    kill -TERM $2
+    wait $2
+    touch $STOP
+    kill -TERM $3
+    wait $3
+    rm $STOP
+}
+
+trap 'before_exit $1 $PID_INSTANCE $PID_WATCH; exit 143' TERM
 
 download_repository $1
-dumb-init learn-ocaml --sync=sync --repo=repository &
-PID=$!
-wait $PID
+learn-ocaml --sync=sync --repo=repository &
+PID_INSTANCE=$!
+
+watch_upload $1 &
+PID_WATCH=$!
```

Comment:
What is the purpose of this variable? It seems not to be used in the rest of the script...

Reply:
The variable get the PID of the watcher process. And I will use it in the trap command of the main process, to terminate it properly